components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -178,12 +183,20 @@ private void checkOrganizationIsAllowedToSwitch(String currentOrgId, String swit
                     "Provided token was already issued for the requested organization.");
         }
         try {
+            if (getOrgApplicationManager().resolveSharedApplication(appName, currentOrgId, switchOrgId) == null) {
```

Comment:
Can't we have the main application's app UUID?
Without loading entier application object of the shared app, we can use https://github.com/wso2-extensions/identity-organization-management/pull/260/files#diff-71616cb5665a5960b04ce296b01925b94d5f6f063bbaf4d45d2c758e94d1f86dR116 method

Reply:
In here we can not get the APP ID. only app name can be find from tokReqMsgCtx

Reply:
- without loading the entier SP, resolve the app id of the application by name (there should be a service) and use the function mentioned above. Then it will avoid loading the shared app during the switch grant flow and main app loading during the login flow.

If you use `resolveSharedApplication` function, internally it will load entier main application object and shared application object as per the current implementation

Shall we expose a similar method as

Reply:
Fixed